bool query
---
LocationArray locationArray
bool result